tests/test_dialects.py

```diff
@@ -2359,3 +2360,68 @@ def test_clickhouse(self):
             read="spark",
             write="clickhouse",
         )
+
```

Comment:
I'm hoping a structure like this (or something similar... ideas welcomed) can help organize these dialect tests.

Still a ton of tests to move here. 

The basic idea is that the bulk of the test suite should be based on parsing/generating to/from the _abstract syntax_. Not to some random other dialect.

We have a separate test method for generate and parsing since not all operations are reversible.

I hope this clarifies the task of adding complete support for a dialect.

Reply:
but then how do you test the reading of dialects? 

i think the structure is ok, but we need to be able to define the read

presto -> spark is the main thing i want to test, as it's an actual use case
presto -> sqlglot is ok, but it's not really end to end

Reply:
what if we created 1 file per dialect, and each dialect always reads that dialect, and then we keep this structure?